Automattic/wp-calypso
# Stepper Onboarding: Fix duplicate calypso_signup_start event

Fixes: https://github.com/Automattic/wp-calypso/issues/95125

## Proposed Changes

* Updates the navigation logic related to `use-my-domain` step.

## Why are these changes being made?


* This fixes the duplicate `calypso_signup_start` event issue. The event duplicated because query params were mistakenly deleted within the onboarding flow. For example, if you load the onboarding flow with `ref` query param, you will notice that the `calypso_signup_start` event is fired and the param will be removed from the URL. When you navigate to another step, the event fires again because the `ref` has changed (in this case its removed).

## Testing Instructions



* Checkout branch or use the live link.
* Navigate to `/setup/onboarding?ref=agTest`.
* Open the Chrome dev tools `Network` tab and filter by `calypso_signup_start`. You may have to refresh the page with the dev tools open to see the event.
* Verify the `ref` param is not removed from the URL when you land on the domains step.
* Click on the `Use a domain I own` link.
* Verify the `calypso_signup_start` event is NOT triggered again. Continue through the flow until checkout to verify you don't see the event triggered again.
* **Testing functionality** - Go back and forth through `domains` -> `use-my-domains` -> `plans` -> checkout to ensure the functionality is still working correctly. Test the browser back button and stepper back button.

## Pre-merge Checklist
